File: Easy/LCE - 942.js
/**
 * @param {string} S
 * @return {number[]}
 */
var diStringMatch = function(S) {
    
    let i=0, d=S.length
    let ans = []
    
    for(x of S) {
        if(x==='I') {
            ans.push(i)
            i++
        }
        if(x==='D') {
            ans.push(d)
            d--
        }
    }
    
    ans.push(i)
    return ans
};

File: Easy/Python - Solutions/LCE_942.py
class Solution:
    def diStringMatch(self, s: str) -> List[int]:
        max = len(s)
        min = 0
        ans = []
        for i in s:
            if i == 'D':
                ans.append(max)
                max -= 1
            else:
                ans.append(min)
                min += 1

        ans.append(min)
        return ans
